pkg/schema/schema.go

```diff
@@ -257,6 +252,147 @@ func Exact(maj, min int) SearchOption {
 	}
 }
 
+// ApplyDefaults returns a new, concrete copy of the Resource with all paths
+// that are 1) missing in the Resource AND 2) specified by the schema,
+// filled with default values specified by the schema.
+func ApplyDefaults(r Resource, scue cue.Value) (Resource, error) {
+	rv, err := rt.Compile("resource", r.Value)
+	if err != nil {
+		return r, err
+	}
+	rvUnified := rv.Value().Unify(scue)
+	re, err := convertCUEValueToString(rvUnified)
+	if err != nil {
+		return r, err
+	}
+	return Resource{Value: re}, nil
+}
+
+func convertCUEValueToString(inputCUE cue.Value) (string, error) {
+	re, err := cuejson.Marshal(inputCUE)
+	if err != nil {
+		return re, err
+	}
+
+	result := []byte(re)
+	result = bytes.Replace(result, []byte("\\u003c"), []byte("<"), -1)
+	result = bytes.Replace(result, []byte("\\u003e"), []byte(">"), -1)
+	result = bytes.Replace(result, []byte("\\u0026"), []byte("&"), -1)
+	return string(result), nil
+}
+
+// TrimDefaults returns a new, concrete copy of the Resource where all paths
+// in the  where the values at those paths are the same as the default value
+// given in the schema.
+func TrimDefaults(r Resource, scue cue.Value) (Resource, error) {
+	rvInstance, err := rt.Compile("resource", r.Value)
+	if err != nil {
+		return r, err
+	}
+	rv, _, err := removeDefaultHelper(scue, rvInstance.Value())
+	if err != nil {
+		return r, err
+	}
+	re, err := convertCUEValueToString(rv)
+	if err != nil {
+		return r, err
+	}
+	return Resource{Value: re}, nil
+}
+
+func isCueValueEqual(inputdef cue.Value, input cue.Value) bool {
+	val, _ := inputdef.Default()
+	err1 := input.Subsume(val)
+	err2 := val.Subsume(input)
+	if err1 == nil && err2 == nil {
+		return true
+	}
+	return false
```

Comment:
Can simplify all this down:
```suggestion
	return input.Subsume(val) == nil && val.Subsume(input) == nil
```